### PR TITLE
earthfile2llb: respect ctx.Done() in withDockerRunRegistry

### PR DESCRIPTION
**Pull request by nelsam**
This watches for ctx.Done() when looping through the res.ResultChan channel from MultiImageSolver.SolveImages when handling a WITH DOCKER RUN command that uses the registry.

### CODE REVIEW

**Review by alexcb — APPROVED** [review]
nice find!

File: earthfile2llb/with_docker_run_reg.go [hunk 1]
```diff
@@ -320,3 +321,18 @@ func (w *withDockerRunRegistry) load(ctx context.Context, opt DockerLoadOpt) (ch
 
 	return imageDefChan, nil
 }
+
+func readImgResults(ctx context.Context, ch chan *states.ImageResult) ([]*states.ImageResult, error) {
+	var results []*states.ImageResult
+	for {
+		select {
+		case result, ok := <-ch:
+			if !ok {
```
Comment (alexcb) [inline earthfile2llb/with_docker_run_reg.go:330]: observation: the expected (non-error) code flow requires this to be "not ok" :laughing:

Comment (alexcb) [reply earthfile2llb/with_docker_run_reg.go:330]: (this isn't even a nit, just an observation I thought was amusing)

Comment (nelsam) [reply earthfile2llb/with_docker_run_reg.go:330]: Hah, that is amusing. It's tempting to use `keepGoing` or something, but I think convention makes it reasonable to still use `ok` in spite of the ... odd flow.

**Pull request merged**